org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+                    {
+                        WriteToLog(string.Format("    Added data to bucket!"));
+                    }
+                    WriteToLog(string.Format("    After Excel: {0}", watch.ElapsedMilliseconds));
+                }
+
+                // Schedule HubSpot update if 1:1 match
+                if ( person != null )
+                {
+                    var properties = new List<HubspotPropertyUpdate>() { new HubspotPropertyUpdate() { property = "rock_person_id", value = person.Id.ToString() } };
+
+                    // If the Hubspot Contact does not have FirstName, LastName, or Phone Number we want to update those...
+                    if ( String.IsNullOrEmpty(contact.properties.firstname) )
+                    {
+                        properties.Add(new HubspotPropertyUpdate() { property = "firstname", value = person.NickName });
+                    }
+                    if ( String.IsNullOrEmpty(contact.properties.lastname) )
+                    {
+                        properties.Add(new HubspotPropertyUpdate() { property = "lastname", value = person.LastName });
+                    }
+                    if ( String.IsNullOrEmpty(contact.properties.phone) )
+                    {
+                        PhoneNumber mobile = person.PhoneNumbers.FirstOrDefault(n => n.NumberTypeValueId == 12);
+                        if ( mobile != null && !mobile.IsUnlisted )
+                        {
+                            properties.Add(new HubspotPropertyUpdate() { property = "phone", value = mobile.NumberFormatted });
+                        }
+                    }
+                    var url = $"https://api.hubapi.com/crm/v3/objects/contacts/{contact.id}";
+                    UpdateHsContact(current_id, url, properties, 0);
+                    WriteToLog(string.Format("    After Request (Id and Props): {0}", watch.ElapsedMilliseconds));
+                }
+                else
+                {
+                    if ( inBucket )
+                    {
+                        //We don't have an exact match but we have guesses, so update Hubspot to reflect that.
+                        var properties = new List<HubspotPropertyUpdate>() { new HubspotPropertyUpdate() { property = "potential_rock_match", value = "True" } };
+                        var url = $"https://api.hubapi.com/crm/v3/objects/contacts/{contact.id}";
+                        UpdateHsContact(current_id, url, properties, 0);
+                        
+                        WriteToLog(string.Format("    After Request (Label): {0}", watch.ElapsedMilliseconds));
+                    }
+                }
+                WriteToLog(string.Format("    End of Iteration: {0}", watch.ElapsedMilliseconds));
+                watch.Stop();
+            }
+
+            // Add contacts for people unknown to HubSpot
+            HashSet<int> hsPersonIds = new HashSet<int>();
+            HashSet<string> hsPersonEmails = new HashSet<string>();
```

Comment:
`HashSet`s aren't meant for this use case. After a quick test, I saw that a `List` is over two times faster for this use case. Can we swap these out for `List`s?

Reply:
Appears these are used for .contains() also